File: src-tauri/src/models/repository.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Repository {
    pub id: String,
    pub owner: String,
    pub name: String,
    pub full_name: String,
    pub local_path: String,
    pub default_branch: String,
    pub is_private: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RepositoryFormData {
    pub owner: String,
    pub name: String,
    pub local_path: String,
}

File: src/types/repository.ts
export type Repository = {
  id: string;
  owner: string;
  name: string;
  full_name: string;
  local_path: string;
  default_branch: string;
  is_private: boolean;
  created_at: string;
  updated_at: string;
};

export type RepositoryFormData = {
  owner: string;
  name: string;
  local_path: string;
  default_branch: string;
  is_private: boolean;
};
